src/fsutil.cpp

```diff
@@ -198,29 +200,59 @@ void chdir(const std::string &directory) {
   }
 }
 
-TempFile::TempFile() {
-  char fname_buf[L_tmpnam];
-  char *s = std::tmpnam(fname_buf);    // gives a warning, but I'm not worried
-  if (s != fname_buf) {
-    throw std::ios_base::failure("Could not create temporary file");
+std::string which(const std::string &command) {
+  const char* path_env = std::getenv("PATH");
+  std::string path;
+  if (path_env != nullptr) {
+    path = path_env;
   }
+  return which(command, path);
+}
 
-  name = fname_buf;
-  name += "-flit-testfile.in";    // this makes the danger much less likely
-  out.exceptions(std::ios::failbit);
-  out.open(name);
+std::string realpath(const std::string &relative) {
+  char* fullpath = ::realpath(relative.c_str(), nullptr);
+  std::string fullpath_string(fullpath);
+  free(fullpath);
+  return fullpath_string;
 }
 
-TempDir::TempDir() {
-  char fname_buf[L_tmpnam];
-  char *s = std::tmpnam(fname_buf);    // gives a warning, but I'm not worried
-  if (s != fname_buf) {
-    throw std::ios_base::failure("Could not find temporary directory name");
+std::string which(const std::string &command, const std::string &path) {
```

Comment:
I feel like this function is inconsistent.

which("/not-a-command", <path>) will return "/not-a-command"
but
which("not-a-command", <path>) will throw a failure

Same goes for relative paths.

I can add the checks, I just want feedback on whether this make sense.

Reply:
Sure, we could do
```c++
if (command[0] == '/') {
  auto status = file_status(command);
  if (status.is_reg) {
    return command;
  }
}
```
does that sound good?

Reply:
The `file_status()` throws a `std::ios_base::failure` if there is no such file or directory, or the current user does not have permission to see the file or folder.

Reply:
I can address this

Reply:
Thanks for the find.  Yes, this should be fixed.  I also found that I have no unit tests for the `which()` function.  That's a problem, too.  I'll do that now.

Reply:
Unit tests added and the `which()` function has been changed to act as expected according to the unit tests.  It is much more consistent now, albeit much longer :dizzy:

Reply:
Awesome! Thank you.